src/lib/gl_engine/tvgGlTessellator.h

```diff
@@ -0,0 +1,184 @@
+
+    bool intersectPairEdge(detail::Edge *left, detail::Edge *right, detail::ActiveEdgeList *ael,
+                           detail::Vertex **current);
+
+    detail::Polygon *makePoly(detail::Vertex *v, int32_t winding);
+
+    void emitPoly(detail::MonotonePolygon *poly);
+
+    void emitTriangle(detail::Vertex *p1, detail::Vertex *p2, detail::Vertex *p3);
+private:
+    FillRule                            fillRule = FillRule::Winding;
+    std::unique_ptr<detail::ObjectHeap> pHeap;
+    Array<detail::VertexList *> outlines;
+    detail::VertexList *pMesh;
+    detail::Polygon    *pPolygon;
+    Array<float>    *resGlPoints;
+    Array<uint32_t> *resIndices;
+};
+
+class Stroker final
+{
+
+    struct State
+    {
+        GlPoint firstPt = {};
+        GlPoint firstPtDir = {};
+        GlPoint prevPt = {};
+        GlPoint prevPtDir = {};
+        bool  hasMove = false;
+    };
+public:
+    Stroker(Array<float> *points, Array<uint32_t> *indices);
+    ~Stroker() = default;
+
+    void stroke(const RenderShape *rshape);
+
+private:
+    void doStroke(const PathCommand *cmds, uint32_t cmd_count, const Point *pts, uint32_t pts_count);
+    void doDashStroke(const PathCommand *cmds, uint32_t cmd_count, const Point *pts, uint32_t pts_count,
+                      uint32_t dash_count, const float *dash_pattern);
+
+    float strokeRadius() const
+    {
+        return mStrokeWidth * 0.5f;
+    }
+
+    void strokeCap();
+
+    void strokeLineTo(const GlPoint &curr);
+
+    void strokeCubicTo(const GlPoint &cnt1, const GlPoint &cnt2, const GlPoint &end);
+
+    void strokeClose();
+
+    void strokeJoin(const GlPoint &dir);
+
+    void strokeRound(const GlPoint &prev, const GlPoint &curr, const GlPoint &center);
+
+    void strokeMiter(const GlPoint &prev, const GlPoint &curr, const GlPoint &center);
+
+    void strokeBevel(const GlPoint &prev, const GlPoint &curr, const GlPoint &center);
+private:
+    Array<float> *mResGlPoints;
+    Array<uint32_t> *mResIndices;
+    float mStrokeWidth = 1.f;
+    float mMiterLimit = 4.f;
+    StrokeCap mStrokeCap = StrokeCap::Square;
+    StrokeJoin mStrokeJoin = StrokeJoin::Bevel;
+    State mStrokeState = {};
+};
+
+class DashStroke
+{
+public:
+    DashStroke(Array<PathCommand> *cmds, Array<Point> *pts, uint32_t dash_count, const float *dash_pattern);
+
+    ~DashStroke() = default;
+
+    void doStroke(const PathCommand *cmds, uint32_t cmd_count, const Point *pts, uint32_t pts_count);
+
+private:
+    void dashLineTo(const GlPoint &pt);
+    void dashCubicTo(const GlPoint &pt1, const GlPoint &pt2, const GlPoint &pt3);
+
+    void moveTo(const GlPoint &pt);
+    void lineTo(const GlPoint &pt);
+    void cubicTo(const GlPoint &pt1, const GlPoint &pt2, const GlPoint &pt3);
+
+private:
+    Array<PathCommand> *mCmds;
+    Array<Point> *mPts;
+    uint32_t mDashCount;
+    const float *mDashPattern;
+    float mCurrLen;
+    int32_t mCurrIdx;
+    bool mCurOpGap;
+    GlPoint mPtStart;
```

Comment:
Just for your information, it doesn't make sense to maintain both GlPoint and tvg::Point. 
Perhaps, we can elevate GlPoint and then integrate it with tvg::Point (in tvgMath.h) through a later refactoring.

Reply:
in draft #1531  I use tvg::Point in tessellator by add some operator function. While in this PR I thought, `sw_engine` use `SwPoint` maybe `gl_engine` needs to use `GlPoint`.

Reply:
SwPoint is a integer type data structure, it's not compatible with tvg::Point, but GLPoint looks compatible. I think we can revise this later.